unskript-ctl/db_utils.py

```diff
@@ -84,11 +86,25 @@ def upsert_pss_record(name: str, data: dict, overwrite: bool=False):
     with db.transaction() as connection: 
         root = connection.root()
         if overwrite:
-            root[name] = data
+            d = json.loads(data)
+            compressed_d = zlib.compress(d.encode('utf-8'))
+            root[name] = json.dumps(compressed_d)
         else:
             d = root[name]
+
+            if isinstance(d, bytes):
+                d = zlib.decompress(d).decode('utf-8')
+
+            if isinstance(d, str):
+                d = json.loads(d)
```

Comment:
shouldnt this be json.dumps???

Reply:
No, here we need it to be in the form of a Dict so we can do a update() on the python dictionary.